src/main/java/org/incendo/serverlib/util/PackageChecker.java

```diff
@@ -0,0 +1,64 @@
+package org.incendo.serverlib.util;
+
+import com.google.common.collect.ImmutableSet;
+
+import java.io.IOException;
+import java.net.JarURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.Enumeration;
+import java.util.Set;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.zip.ZipEntry;
+
+public final class PackageChecker {
+    private static final Logger LOGGER = Logger.getLogger(PackageChecker.class.getName());
+
+    private PackageChecker() {
+    }
+
+    private static final Set<String> KNOWN_ENTRIES;
+
+    static {
+        final ImmutableSet.Builder<String> builder = ImmutableSet.builder();
+        try {
+            final Enumeration<URL> urls = ClassLoader.getSystemClassLoader().getResources("META-INF");
+            while (urls.hasMoreElements()) {
+                final URL url = urls.nextElement();
+                final URLConnection urlConnection = url.openConnection();
+                if (!(urlConnection instanceof JarURLConnection)) {
+                    continue;
+                }
+                final JarURLConnection connection = (JarURLConnection) urlConnection;
+                try (final JarFile jar = connection.getJarFile()) {
+                    final Enumeration<JarEntry> entries = jar.entries();
```

Comment:
No checks are made to only keep packages. Perhaps that would be beneficial, as otherwise this will be a huuuge set of all classes where only packages are necessary?